### ISSUE TITLE
Cloud app brings iotivity-constrained based OCF servers down

### ISSUE THREAD

**Issue by uhofemei** [01org/SmartHome-Demo#140]
I have observed that launching the cloud web portal, regardless whether it was running on the gateway itself or on a different host on the network, will cause all iotivity-constrained based OCF serves down within <= 2 minutes. This is true for the native multi sensor Arduino or the ZJS solution.  James Prestwood noted that a proper JSON syntax (formatting, order, and number of arguments) the sensor is accessed with is very important. Wrong JSON syntax for a given sensor will bring the resource down. I suspect this is what we are dealing with here.

@ichbinblau, I think to incorporate OCF support for micro controllers, such as the Arduino 101, will require a review of how the Python celery service that is using the REST API is accessing the sensor. This is true for the JSON syntax used for each sensor, as well as the frequency at which each sensor is polled. I will send you an email with login credentials for the gateway and the "fog" device, which is running the cloud docker container now.

**Comment by ttzeng** [comment]
@ichbinblau , while investigating this issue, I found once this issue occurs, even I restart the ZJS servers, the cloud portal won't be able to get the updated report from the API server. The UUID of ZJS servers will change on every reboot, but it looks like the cloud portal continuously polling the dead server:

```
[2017-05-17 09:44:24,659: INFO/PoolWorker-6] Starting new HTTP connection (1): 10.5.221.145
[2017-05-17 09:44:29,665: INFO/PoolWorker-6] CeleryTask.tasks.get_sensor_data[288cc31c-bdde-4b7c-b188-5a7405213eaf]: running ids: [], total: 0
[2017-05-17 09:44:29,665: INFO/PoolWorker-6] CeleryTask.tasks.get_sensor_data[288cc31c-bdde-4b7c-b188-5a7405213eaf]: dead ids: set([])
[2017-05-17 09:44:29,665: INFO/PoolWorker-6] CeleryTask.tasks.get_sensor_data[288cc31c-bdde-4b7c-b188-5a7405213eaf]: active threads: [], total: 0
[2017-05-17 09:44:39,668: INFO/PoolWorker-6] http://10.5.221.145:8000/api/oic/res
```
The API server has discovered the rebooted server's UUID 5ae36352-47e9-40d8-b966-d62884434c5b, but the cloud portal is still expecting the dead UUID 288cc31c-bdde-4b7c-b188-5a7405213eaf. I have tried CTRL-F5 to clear the cache, and restart the container, the cloud portal never polling the new UUID. Any idea why the new UUID never picked by the cloud portal?

**Comment by ichbinblau** [comment]
Hi, @ttzeng.  It looks like the resource table has not been synced up with the API server. It means that the celery worker did not get updated on the changes from the API server. It is still using old UUID and resource list even though the container was restarted. 
How about remove the docker container and run it again?  In that case, the resource table will be completely removed and created again. If it works, there might be some problems on updating the table in code and I need more info to debug it. Otherwise, it is probably due to the connectivity to the API server. 
Would you plz have a try?

**Comment by uhofemei** [comment]
How can I run the web portal services without docker?

**Comment by ttzeng** [comment]
@ichbinblau , remove the container and run a fresh cloud portal can reach to the new OCF server, but once the server restart, it will get new UUID, and even the API server can report the new UUID, the cloud portal looks like unable to get the updates from the API server. Any idea?

**Comment by ichbinblau** [comment]
@uhofemei under the project folder, there is a wiki on how to deploy web portal on Ubuntu 14.04. I think it should mostly apply to v16.04.

**Comment by gvancuts** [comment]
@ichbinblau , should we update the wiki page to indicate Python-2.7.12 (that's what we use in the Dockerfile) should be used instead of 2.7.10 (currently documented in the wiki).

What about the TAP-related instructions [here](https://github.com/01org/SmartHome-Demo/wiki/How-to-set-up-the-web-portal-in-OpenStack%3F#configure-the-web-settings), how should these be treated in the case we're not running on TAP?

**Comment by uhofemei** [comment]
@ichbinblau, I'll be in the lab at 5 PM tonight and you can poke at things. I will restart the MCU for you. Sound good?

**Comment by ichbinblau** [comment]
@gvancuts I have updated the python version in the wiki and removed the TAP-related configuration as we would no longer use TAP. The configuration is to get the total number of instances scaled up in the TAP environment. If we are using container, the instance id and total number will be N/A.

**Comment by ichbinblau** [comment]
@uhofemei I am good with the time. I will ping you if I get up earlier.

**Comment by ichbinblau** [comment]
Hi, @ttzeng. I have no idea why it happens. Would you keep the env (the new UUID cannot be picked) and grant me the access to take a look?

**Comment by uhofemei** [comment]
I also filed a bug in the ZJS issue tracker. 01org/zephyr.js#1237

Geoff was telling me they will re-direct this to Kishen for further investigation. The isuse opened in the iotivity-constrained repo that discusses this is: iotivity/iotivity-constrained#5

**Comment by gvancuts** [comment]
> I also filed a bug in the ZJS issue tracker. 01org/zephyr.js#1237

Thank you @uhofemei

> Geoff was telling me they will re-direct this to Kishen for further investigation. The isuse opened in the iotivity-constrained repo that discusses this is: iotivity/iotivity-constrained#5

I don't see the link between our issue and that one?

**Comment by uhofemei** [comment]
I agree there seems to be a disconnect. The issue being that iotivity-constrained won't build with Zephyr 1.8 RC1 certainly affects us, but I think, I will file a new issue based on ZJS #1237.
